src/main/java/com/google/devtools/build/lib/remote/http/HttpCacheClient.java

```diff
@@ -774,6 +808,19 @@ public void close() {
     return null;
   }
 
+  private File createCompressedUpload(String name, InputStream input) throws IOException {
+    File compressedOutput = File.createTempFile(name, ".gz");
+    compressedOutput.deleteOnExit();
+    OutputStream out = new DeflaterOutputStream(new FileOutputStream(compressedOutput));
+    byte[] buffer = new byte[131072];
+    int len;
+    while ((len = input.read(buffer)) > 0) {
+      out.write(buffer, 0, len);
+    }
```

Comment:
Nit: can be simplified with `ByteStreams.copy`

Reply:
Doesn't say what buffer size it uses, prefer doing it this way.